spec/unit/util_spec.rb

```diff
@@ -233,7 +233,15 @@ def get_mode(file)
       # This is ugly, but I can't really think of a better way to do it without
       # letting it actually close fds, which seems risky
       (0..2).each {|n| IO.expects(:new).with(n).never}
-      (3..256).each {|n| IO.expects(:new).with(n).returns mock('io', :close) }
+
+      fds = Dir.entries("/proc/self/fd")
```

Comment:
You could also mock Dir access to '/proc/self/fd' to return a fixed set of files.